utils/common.py

```diff
@@ -124,20 +125,22 @@ def get_valid_xml_string(string, escape=True):
     return ''
 
 
-def format_date(date):
-    return date and date.strftime('%d-%m-%Y')
+def deep_date_format(date: Union[datetime.date, datetime.datetime], fallback: Optional[str] = '') -> Optional[str]:
+    if date:
```

Comment:
This statement implies that the date can be `None`. However the date argument signature says it cannot be `None`. Need to add `None` inside the Union.

Reply:
Done